toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -33,6 +34,99 @@ func PrintBuildResult(res *BuildResult) {
 	}
 }
 
+// RecordBuildSummary stores the summary in to a csv.
+func RecordBuildSummary(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, buildState *GraphBuildState, outputPath string) {
```

Comment:
This function and `PrintBuildSummary` seem to do a lot of the same work, with regards to processing the build graph. Can we refactor out the common graph processing steps, output extracted data to a common data structure, and have print/record functionality work off of the extracted data?

Reply:
It will make the for loop run once instead of twice. When the entire build time is in hours, optimizing this for loop provides negligible performance advantage. Let me know if you think otherwise.